### PR TITLE
Filter blocked bugs

### LINKED ISSUE #39: Filter blocked bugs out from "To Fix" tab
I can't fix them since they are blocked!

### PR DESCRIPTION
**Pull request by fitzgen**
Fixes #39.

### CODE REVIEW

File: app/fixes.js [hunk 1]
```diff
@@ -1,5 +1,20 @@
 function Fixes() {}
 Fixes.prototype = new Queue();
+
+try {
+  Fixes.prototype.includeBlockedBugs = JSON.parse(localStorage['bztodos-include-blocked-bugs']);
+} catch (e) {
```
Comment (harthur) [inline app/fixes.js]: As little as possible in the `try/catch`, just the bare minimum of what is okay to throw.

File: app/queue.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 function Queue() {
-  this.items = [];
+  this._items = [];
```
Comment (harthur) [inline app/queue.js]: This ain't Moz code! I don't have underscore-prefixed properties in this project.

File: app/user.js [hunk 3]
```diff
@@ -271,17 +274,62 @@ User.prototype.toFix = function(callback) {
...
+            return { bug: bug };
+         })
+         callback(null, bugsToFix);
+      });
    });
 }
 
+// Fetch all of each bugs dependencies and modify in place each bug's depends_on
+// array so that it only contains OPEN bugs that it depends on.
+User.prototype._fetchDeps = function(bugs, callback) {
```
Comment (harthur) [inline app/user.js]: no underscores here.

File: app/user.js [hunk 4]
```diff
@@ -271,17 +274,62 @@ User.prototype.toFix = function(callback) {
...
+   var waiting = 0;
+
+   function maybeFinish() {
```
Comment (harthur) [inline app/user.js:298]: Explain what this is and why we need to do this.

Hm, can't we use events here?

**Comment by harthur** [comment]
This causes a weird style issue, where the dates are pushed onto the next line:

screen shot 2013-09-19 at 4 03 37 pm

**Comment by fitzgen** [comment]
> This causes a weird style issue, where the dates are pushed onto the next line:

Not caused by my patch. Seeing this CSS bug on bugmotodo.org, as well.

**Comment by fitzgen** [comment]
> Hm, can't we use events here?

Too heavy weight, IMO; this is just a simple little helper. Hope the added comments clear things up.

File: app/user.js [hunk 5]
```diff
@@ -271,17 +274,58 @@ User.prototype.toFix = function(callback) {
...
+      }
+
+      var oldDeps = bug.depends_on;
+      bug.depends_on = [];
+      oldDeps.forEach(function(dep) {
+         waiting++;
+         self.client.getBug(dep, function (err, depBug) {
+            waiting--;
+            try {
+               if (err) return console.error("Error fetching bug " + dep);
```
Comment (harthur) [inline app/user.js]: File has braces and newlines around if statements:

``` javascript
if (err) {
  return console.error("error");
}
```

File: app/user.js [hunk 6]
```diff
@@ -271,17 +274,58 @@ User.prototype.toFix = function(callback) {
...
+               if (depBug.status === "RESOLVED") return;
+               bug.depends_on.push(depBug);
+            } finally {
```
Comment (harthur) [inline app/user.js:321]: What is the `try...finally` for?

**Comment by fitzgen** [comment]
Lots of lines without braces + newlines of the form:

```
if (err) throw err;
```

Would you prefer that here?

File: app/fixes.js [hunk 7]
```diff
@@ -12,4 +31,19 @@ Fixes.prototype.fetch = function() {
...
+
+FixList.prototype.initialize = function(collection, type) {
+  QueueList.prototype.initialize.call(this, collection, type);
+
+  this.checkbox = $("#include-blocked-bugs");
+  this.checkbox.attr("checked", this.collection.includeBlockedBugs);
+  this.checkbox.change(this._setIncludeBlockedBugs.bind(this));
+};
+
+FixList.prototype._setIncludeBlockedBugs = function(event) {
```
Comment (harthur) [inline app/fixes.js]: no underscore in function name.

**Comment by fitzgen** [comment]
> What is the `try...finally` for?

To make sure that you call `maybeFinish` (and therefore the callback) no matter which branch of logic you take. Will add comment.

> no underscore in function name.

Woops, forgot that one.

**Comment by harthur** [comment]
"Lots of lines without braces + newlines of the form:

```
if (err) throw err;
```

Would you prefer that here?"

Oh, so there is, sorry to be confusing. I guess I thought those were short enough /=

**Pull request merged**